### PR TITLE
Event toggle function pause new

### PR DESCRIPTION
**Pull request by smbsp**
closes: #297
Added an event ToggleFunctionPaused to the function toggleFunctionPause()

We earlier had Contract Size issues and hence this version is built using updated Loan Token logic using Beacon Chain. This PR is a replacement of: https://github.com/DistributedCollective/Sovryn-smart-contracts/pull/317

### CODE REVIEW

File: contracts/connectors/loantoken/modules/LoanTokenSettingsLowerAdmin.sol [hunk 1]
```diff
@@ -193,9 +195,14 @@ contract LoanTokenSettingsLowerAdmin is LoanTokenLogicStorage {
 					uint256(0xd46a704bc285dbd6ff5ad3863506260b1df02812f4f857c8cc852317a6ac64f2)
 				)
 			);
+		assembly {
+			paused := sload(slot)
+		}
+		require(paused != isPaused, "invalid");
```
Comment (ororopickpocket) [inline contracts/connectors/loantoken/modules/LoanTokenSettingsLowerAdmin.sol:201]: the require is not strictly required here, but doesn't hurt either. however, the error message doesn't provide any valuable information. change it to "isPaused is already set to that value" or similar message.

File: contracts/connectors/loantoken/modules/LoanTokenSettingsLowerAdmin.sol [hunk 2]
```diff
@@ -193,9 +195,14 @@ contract LoanTokenSettingsLowerAdmin is LoanTokenLogicStorage {
...
 		assembly {
 			sstore(slot, isPaused)
 		}
+		emit ToggleFunctionPaused(funcId, !isPaused, isPaused);
```
Comment (ororopickpocket) [inline contracts/connectors/loantoken/modules/LoanTokenSettingsLowerAdmin.sol:205]: the event should rather be called
ToggledFunctionPause
(toggle is the verb here)

**Review by ororopickpocket — CHANGES_REQUESTED** [review]

File: contracts/connectors/loantoken/modules/LoanTokenSettingsLowerAdmin.sol (hunk 1 above)
Comment (smbsp) [reply contracts/connectors/loantoken/modules/LoanTokenSettingsLowerAdmin.sol]: Done

File: contracts/connectors/loantoken/modules/LoanTokenSettingsLowerAdmin.sol (hunk 2 above)
Comment (smbsp) [reply contracts/connectors/loantoken/modules/LoanTokenSettingsLowerAdmin.sol]: Done

**Review by ororopickpocket — APPROVED** [review]

**Review by tjcloa — APPROVED** [review]

**Review by Alitasovryn — COMMENTED** [review]
Looks fine.

**Pull request merged**